packages/styles/scss/components/masthead/_masthead-megamenu.scss

```diff
@@ -123,13 +132,109 @@
+       * the stateful information needed to dictate how elements in the lower shadow
+       * roots should appear, and we can pass that information down via custom props.
+       */
+      &.#{$prefix}--masthead__megamenu__container--row--tab {
         @include carbon--make-row;
 
-        background-color: $ui-03;
         flex-wrap: nowrap;
         gap: 1px;
+
+        ::slotted(#{$dds-prefix}-megamenu-right-navigation) {
+          --#{$dds-prefix}-right-nav-padding: #{$layout-01};
+        }
+      }
+
+      &.#{$prefix}--masthead__megamenu__container--row--list {
+        @include carbon--breakpoint-down(lg) {
+          @include masthead-megamenu-limit-height;
+          --#{$dds-prefix}-view-all-inset: calc(#{$layout-01} * -1);
+          --#{$dds-prefix}-view-all-container-position: initial;
+          --#{$dds-prefix}-view-all-container-transform: initial;
+
+          // Set containing block for fixed position view all CTA.
+          transform: rotate(0deg);
+          position: relative;
+          z-index: 1;
+
+          margin-inline: calc(#{$layout-04} * -1);
+        }
+
+        @include carbon--breakpoint(lg) {
+          @include carbon--make-row;
+
+          flex-wrap: nowrap;
+          gap: 1px;
+        }
+
+        ::slotted(#{$dds-prefix}-megamenu-left-navigation),
+        ::slotted(#{$dds-prefix}-megamenu-right-navigation) {
+          @include carbon--breakpoint-down(lg) {
+            --#{$dds-prefix}-view-all-padding: #{$layout-01};
+            --#{$dds-prefix}-masthead-max-height: initial;
+            --#{$dds-prefix}-masthead-overflow: initial;
+
+            width: 100%;
+            max-width: initial;
+            margin-inline: initial;
+            padding-inline: $layout-03;
+            border-bottom: 1px solid $ui-03;
+            overflow: initial;
+            max-height: initial;
+
+            .#{$prefix}--masthead__megamenu-container {
+              max-height: initial;
+              overflow: initial;
+            }
+          }
+        }
+
+        ::slotted(#{$dds-prefix}-megamenu-left-navigation) {
+          --#{$dds-prefix}-left-nav-columns: 2;
+
+          @include carbon--breakpoint(lg) {
+            --#{$dds-prefix}-left-nav-columns: initial;
+          }
+        }
+
+        ::slotted(#{$dds-prefix}-megamenu-right-navigation) {
+          --#{$dds-prefix}-right-nav-padding: 0;
+
+          @include carbon--breakpoint-down(lg) {
+            --#{$dds-prefix}-view-all-display: none;
+
+            border-bottom: none;
+          }
+
+          @include carbon--breakpoint(lg) {
+            --#{$dds-prefix}-right-nav-padding: #{$layout-01};
+          }
+        }
+
+        ::slotted([part*='view-all']) {
+          --#{$dds-prefix}-view-all-padding: #{$layout-04};
+
+          position: sticky;
+          z-index: 1;
+          inset-block-end: 0;
+          background-color: $ui-background;
+          border-top: 1px solid $ui-03;
```

Comment:
Same comment - https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/10190/files#r1134581317